components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -435,6 +435,14 @@ void RewardsServiceImpl::CheckPreferences() {
     // utility process.
     StartEngineProcessIfNecessary();
   }
+
+  // If Rewards is enabled and the user has a linked account, then ensure that
+  // the "search result ads enabled" pref has the appropriate default value.
+  if (prefs->GetBoolean(prefs::kEnabled) &&
+      !prefs->GetString(prefs::kExternalWalletType).empty() &&
+      !prefs->HasPrefPath(brave_ads::prefs::kOptedInToSearchResultAds)) {
+    prefs->SetBoolean(brave_ads::prefs::kOptedInToSearchResultAds, false);
+  }
```

Comment:
@zenparsing this needs more thought, so I will discuss this with @aseren before we merge. Thanks

Reply:
This looks good to us.